package.json

```diff
@@ -6,7 +6,7 @@
     "access": "public"
   },
   "scripts": {
-    "build": "webpack",
+    "build": "node scripts/support_node_17+.js webpack",
```

Comment:
Are we able to update the `.node-version` file to the latest LTS version?

Reply:
Yes definitely! I can remove the node scripts since we don't need to use `--openssl-legacy-provider` anymore. Note: nvm does not support reading a `.node-version` file.

Reply:
> Yes definitely! I can remove the node scripts since we don't need to use --openssl-legacy-provider anymore.

Awesome! Thank you.

> Note: nvm does not support reading a .node-version file.

Yup, unfortunately there's no standard on node version lock files ([see .node-version file usage](https://github.com/shadowspawn/node-version-usage)). The node version manager I use, [fnm](https://github.com/Schniz/fnm), does support it. If you like `nvm`, feel free to add the `.nvmrc` file as well!

Reply:
@alonso-cadenas I dug a little bit deeper into understanding why nvm doesn't support `.node-version`, and I believe that the nvm team does not consider it to be standard tooling (https://github.com/nvm-sh/nvm/issues/794). What are your thoughts on removing it and relying on `package.json#engines` like in https://github.com/robhogan/react-native/commit/f114ad0d26d445b5d154cff7c59ae39163e83390?

Reply:
TLDR: I suggest we add the `engines` prop to our package.json and keep/add any node version lock files that are useful to you and I for development, including `.node-version`, `.nvmrc`, etc.

I'll present my reasoning by answering the key points in the react-native commit you shared:

> IMO, there's no value in having `.node-version` files either in the repository or the template.

It's valuable to me for development because using a node version manager allows me to install multiple node versions locally so that I may run multiple applications that each work on specific node versions, or in the case of this SDK, I can make sure it works on multiple versions.

Furthermore, using a node version lock file makes using a node version manager even easier because it will use that file to automatically detect the ideal node version of a project, so I don't have to run any CLI commands to switch node versions. It removes the question "am I running the right version?" in case the project build fails.

> I don't believe it's standard practice

Correct, there is no standard node version lock file, but it is standard industry practice to use lock files for versioning (npm, yarn, Ruby, etc) - [even nvm supports version lock files with .nvmrc](https://github.com/nvm-sh/nvm#nvmrc). It's actually weird that Node.js doesn't have an official version manager.

> the support from tooling is patchy (at least `nvm` [ignores it](https://github.com/nvm-sh/nvm/issues/794)

It looks like [most node managers actually support the .node-version file](https://github.com/shadowspawn/node-version-usage#node-version-file-usage), with `nvm` being a defector with their .nvmrc file.

> Node is often used without a version manager

It'd be interesting to see some data to back this up since this hasn't been my experience with colleagues. Either way, @esme and I are the only developers contributing to this repo so this is really about what makes our dev experience easier.

> Libraries should use `package.json#engines` to specify their Node JS compatibility ranges
Absolutely, but this property accomplishes a different goal from node version files. It specifies the Node.js engines that our SDK package works on. Since we don't have that property in our package.json at this time, we're claiming our package supports all Node.js versions. We should probably make sure this claim is accurate (which means we'll use our node version manager to test it with all Node.js versions available haha).

My last point is "does it hurt to have a .node-version file in there?". It doesn't look like it does. If you're using a version manager that supports it, then great! If you're an nvm user, or using a version manager that doesn't support `.node-version`, then the node version manager will simply ignore the file.

Reply:
These are valid points - thank you for explaining! My main concern here is having the version in multiple `.node-version` and `.nvmrc` files along with `engines` which could be confusing for someone else contributing to the repo or even using the package in their app.

I agree that it is useful for testing multiple versions, and I think we could develop with the .node-version file locally and gitignore it. This would ensure compatibility with any preferred node version manager as well.

Reply:
Alrighty! Feel free to make your proposed changes 😄